Sources/TuistCore/Graph/GraphTraverser.swift

```diff
@@ -1605,4 +1588,8 @@ extension GraphDependency {
             return nil
         }
     }
+
+    fileprivate var isPrecompiledAndLinkable: Bool {
+        xcframeworkDependency != nil || isPrecompiledDynamicAndLinkable
```

Comment:
When `xcframeworkDependency` is nil, it doesn't necessarily mean it's precompiled _or_ linkable. You will need to adjust the checks to be truthful to the variable name.

The target is precompiled when it's an xcframework that has an xcframework dependency (we check the latter only).

Reply:
> When xcframeworkDependency is nil

my expression checks if `xcframeworkDependency` is **not** nil

> The target is precompiled when it's an xcframework that has an xcframework dependency (we check the latter only).

As far as I understood this sentence, you wrote exactly what `xcframeworkDependency` [getter does](https://github.com/tuist/tuist/pull/7006/files#diff-e92bb6ba816dd394aacf7d74e51dc2269fc21f4b2f1f27031ffede91837cb0e5R1584-R1589).

Reply:
`isPrecompiledDynamicAndLinkable` is declared in the [XcodeGraph](https://github.com/tuist/XcodeGraph) repository. It seems like we shouldn't change it there.
There are two places it's needed as implemented in the `XcodeGraph`.
`isPrecompiledAndLinkable` is used one time in my PR

Reply:
> expression checks if xcframeworkDependency is not nil

You are right – but I find it doesn't read well. What do you think about changing this to:
```suggestion
        switch self {
          case .xcframework: true
          default: isPrecompiledDynamicAndLinkable
        }
```